07_errors/nickolee/storer_test.go

```diff
@@ -0,0 +1,76 @@
+package puppystorer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type StorerSuite struct {
+	suite.Suite
+	store Storer
+}
+
+// A bunch of seed puppies
+var puppies = []*Puppy{
+	{Breed: "Laika", Colour: "First dog on moon", Value: 2130.50},
+	{Breed: "Pomsky", Colour: "Violet", Value: 777.77},
+	{Breed: "Dalmatian", Colour: "Cruella de vil", Value: 101},
+	{Breed: "Rare Lvl 100 Arcanine", Colour: "Brown", Value: 9999.99},
+	{Breed: "Cheap Pug", Colour: "Cheap Brown", Value: -500.50},
+}
+
+func (suite *StorerSuite) TestCreateAndRetrievePuppy() {
+	puppyID, err := suite.store.CreatePuppy(puppies[3])
+	suite.Nil(err)
+	puppy, err := suite.store.ReadPuppy(puppyID)
+	suite.Nil(err)
+	suite.Equal(puppy, puppies[3])
+
+	// Test negative value puppy
+	_, err = suite.store.CreatePuppy(puppies[4])
+	suite.Equal("Error code 400: Sorry puppy value cannot be negative. The dog has to be worth something :)",
```

Comment:
You should check the error and its values specifically
```go
_, rawErr := suite.store.CreatePuppy(puppies[4])
suite.Require().TypeIs(&Error{}, rawErr) // Require will stop the test if error is wrong type, preventing a panic on the next line
err, _ := rawErr.(*Error) // Type cast, err now holds the actual error
suite.Equal(ErrNegativePuppyID, err.Code)
suite.Equal(expectedMessage, err.Message)
```

Reply:
Got it. Thanks for the tip. Was able to wrap my head around typecasting thanks to this!